docs/developer-guide/nodejs-api.md

```diff
@@ -130,13 +130,18 @@ The return value is an object containing the results of the linting operation. H
                     column: 23,
                     message: "Expected a semicolon."
                 }
-            ]
+            ],
+            errorCount: 1,
+            warningCount: 0
         }
-    ]
+    ],
+    errorCount: 1,
+    warningCount: 0
 }
 ```
 
-The top-level report object has a `results` array containing all linting results for files that had warnings or errors (any files that did not product a warning or error are omitted). Each file result includes the `filePath` and a `messages` array. The `messages` array contains the result of calling `linter.verify()` on the given file.
+The top-level report object has a `results` array containing all linting results for files that had warnings or errors (any files that did not product a warning or error are omitted). Each file result includes the `filePath`, a `messages` array, `errorCount` and `warningCount`. The `messages` array contains the result of calling `linter.verify()` on the given file. The `errorCount` and `warningCount` gives the exact number of errors and warning respectively on the given file.
```

Comment:
- "did not **product** a warning" => "did not **produce** a warning"
- "`errorCount` and `warningCount` **gives** the exact" => "`errorCount` and `warningCount` **give** the exact" (verb agreement with the compound subject)
- "exact number of errors and **warning** on the given" => "exact number of errors and **warnings** on the given"